src/parser/select-parser.ts

```diff
@@ -76,6 +77,8 @@ type ExtractAliasFromSelectExpression<SE> = SE extends string
   ? QA
   : SE extends DynamicReferenceBuilder<infer RA>
   ? ExtractAliasFromStringSelectExpression<RA>
+  : SE extends AliasedAggregateFunctionBuilder<any, any, any, infer QA>
```

Comment:
Don't you also need the callback version here?

We need to be careful to not add too many types to these types. The next time a new type needs to be added here, maybe we can think of ways to combine some of these.

Reply:
> Don't you also need the callback version here?

Good catch! done.

> We need to be careful to not add too many branches to these types. The next time a new type needs to be added here, maybe we can think of ways to combine some of these.

Agreed!

Is this a maintainability concern and/or typescript stability concern?

Reply:
At least maintainability, maybe typescript speed/stability too. Haven't run any tests on that.